api/src/logging/flask_logger.py

```diff
@@ -154,14 +154,17 @@ def _add_global_context_info_to_log_record(record: logging.LogRecord) -> bool:
 
 
 def _get_request_context_info(request: flask.Request) -> dict:
+    internal_request_id = str(uuid.uuid4())
+    flask.g.internal_request_id = internal_request_id
+
     data = {
         "request.id": request.headers.get("x-amzn-requestid", ""),
         "request.method": request.method,
         "request.path": request.path,
         "request.url_rule": str(request.url_rule),
-        # A backup ID in case the x-amzn-requestid isn't passed in
```

Comment:
We should make sure we don't lose track of this, either using the X-amzn one in place of this or as an additional identifier (when it's present, and falling back to one we generate when it's missing). But we don't have to change that in this PR, if we want to just open a ticket to follow up in near future Sprint.

Reply:
We'll have to see whether the x-amzn-requestid field is even populated, that might have been something specific to the past project that this originally came from. I just left that one until we have a chance to dig into it more